File: go/longest-subarray-of-1s-after-deleting-one-element.go
func longestSubarray(nums []int) int {
	prevSegmentSize := 0
	currentSegmentSize := 0
	longestSize := 0
	isThereZeros := false
	for _, num := range nums {
		if num == 1 {
			currentSegmentSize += 1
		} else {
			candidateSize := prevSegmentSize + currentSegmentSize
			if longestSize < candidateSize {
				longestSize = candidateSize
			}
			prevSegmentSize = currentSegmentSize
			currentSegmentSize = 0
			isThereZeros = true
		}
	}
	candidateSize := prevSegmentSize + currentSegmentSize
	if longestSize < candidateSize {
		longestSize = candidateSize
	}
	if isThereZeros == false {
		longestSize -= 1
	}
	return longestSize
}

File: python/longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        prev_segment_size = 0
        current_segment_size = 0
        longest_size = 0
        is_there_zeros = False
        for num in nums:
            if num == 1:
                current_segment_size += 1
            else:
                candidate_size = prev_segment_size + current_segment_size
                longest_size = max(longest_size, candidate_size)
                prev_segment_size = current_segment_size
                current_segment_size = 0
                is_there_zeros = True
        candidate_size = prev_segment_size + current_segment_size
        longest_size = max(longest_size, candidate_size)
        if is_there_zeros == False:
            longest_size -= 1
        return longest_size
